### PR TITLE
Close handling

### PR DESCRIPTION
**Pull request by felix-re**
add status code checks for closing_frames and some minor bugfixes

### CODE REVIEW

File: src/websocket.c [hunk 1]
```diff
@@ -179,11 +188,15 @@ static enum websocket_callback_return ws_handle_frame(struct websocket *s, uint8
 		break;
 
 	case WS_CLOSE_FRAME: {
-		uint16_t status_code = 0;
+		uint16_t status_code = 1000;
```
Comment (gatzka) [inline src/websocket.c]: Please use the enum ws_status_code, no magic values.

File: src/websocket.c [hunk 2]
```diff
@@ -63,6 +63,15 @@ static void unmask_payload(uint8_t *buffer, size_t length, uint8_t *mask)
 	}
 }
 
+static bool is_status_code_invalid(uint16_t status_code)
+{
+	bool ret = true;
+	if ((status_code >=1000) && (status_code <= 1003)) ret = false;
```
Comment (gatzka) [inline src/websocket.c]: Please use and enhance the enum ws_status_code.

File: src/websocket.c [hunk 3]
```diff
@@ -182,8 +182,13 @@ static enum websocket_callback_return ws_handle_frame(struct websocket *s, uint8
 	}
 
 	case WS_PONG_FRAME:
-		if (s->pong_received != NULL) {
-			ret = s->pong_received(s, frame, length);
+		if (unlikely(length > 125)) {
```
Comment (gatzka) [inline src/websocket.c]: Would be great to avoid the magic value by a const like WS_SMALL_FRAME_SIZE.

**Review by gatzka — CHANGES_REQUESTED** [review]

File: src/websocket.c (hunk 1 above)
Comment (felix-re) [reply src/websocket.c]: all done

File: src/websocket.c [hunk 4]
```diff
@@ -56,6 +56,8 @@ static const uint8_t WS_HEADER_FIN = 0x80;
 #define WS_PING_FRAME 0x9
 #define WS_PONG_FRAME 0x0a
 
+#define WS_SMALL_FRAME_SIZE 125
```
Comment (gatzka) [inline src/websocket.c]: If possible use static const unsigned int instead of a define. This leads to a symbol which is helpful while debugging.

**Review by gatzka — CHANGES_REQUESTED** [review]

**Review by gatzka — APPROVED** [review]

**Comment by felix-re** [comment]
done? :smile:

**Pull request merged**